File: programmers/2021 KAKAO BLIND RECRUITMENT/합승 택시 요금/solution.js
function solution(n, s, a, b, fares) {
  const INF = Number.MAX_SAFE_INTEGER
  let answer = INF

  const costs = Array.from({ length: n }, (_, i) => {
    let temp = new Array(n).fill(INF)
    temp[i] = 0
    return temp
  })
  fares.forEach(([from, to, cost]) => {
    costs[from - 1][to - 1] = cost
    costs[to - 1][from - 1] = cost
  })

  for (let by = 0; by < n; by++)
    for (let from = 0; from < n; from++)
      for (let to = 0; to < n; to++)
        costs[from][to] = Math.min(
          costs[from][to],
          costs[from][by] + costs[by][to]
        )

  for (let middle = 0; middle < n; middle++)
    answer = Math.min(
      answer,
      costs[s - 1][middle] + costs[middle][a - 1] + costs[middle][b - 1]
    )

  return answer
}

File: programmers/2021 KAKAO BLIND RECRUITMENT/합승 택시 요금/solution.py
def solution(n, s, a, b, fares):
    fare = [[0 if node_from == node_to else float('inf') for node_to in range(n)] for node_from in range(n)]

    for [c, d, f] in fares:
        fare[c-1][d-1] = fare[d-1][c-1] = f

    for node_by in range (n):
        for node_from in range(n):
            for node_to in range(n):
                fare[node_from][node_to] = min(fare[node_from][node_to], fare[node_from][node_by] + fare[node_by][node_to])

    return min(fare[s - 1][middle] + fare[middle][a - 1] + fare[middle][b - 1] for middle in range(n))
